cmd/soroban-rpc/internal/methods/simulate_transaction.go

```diff
@@ -96,15 +98,22 @@ func NewSimulateTransactionHandler(logger *log.Entry, ledgerEntryReader db.Ledge
 			}
 		}
 
+		hostFunctionResults := make([]SimulateHostFunctionResult, len(result.Results))
+		for i := 0; i < len(hostFunctionResults); i++ {
+			hostFunctionResults[i].XDR = result.Results[i]
+		}
+
+		// For now, attribute the full auth and and events to the first function
+		//
+		// FIXME: this is wrong! we should be able to get the auth and events for each separate function
+		//        but needs to be implemented in libpreflight first
+		hostFunctionResults[0].Auth = result.Auth
```

Comment:
for consideration, maybe set each `hostFunctionResults[n].Auth = result.Auth` for now to just set in place the expectations on model, upstream clients can code to that, later when libpreflight can support Auth per host function invocation then upstream doesn't need to roll new version for parsing any different.